packages/at_secondary_server/lib/src/server/at_secondary_config.dart

```diff
@@ -675,6 +688,21 @@ class AtSecondaryConfig {
     }
   }
 
+  static List<String> get protectedKeys {
+    try {
+      YamlList keys = getConfigFromYaml(['hive', 'protectedKeys']);
```

Comment:
I'd like to see unit tests with various flavours of yaml config to verify the logic

One way of enabling those tests to be written would be to augment AtSecondaryConfig to have a way to inject some yaml and change its `getConfigFromYaml` to use that yaml if it is set

Reply:
Changes to allow config injection and the test itself are in [8ee7c62](https://github.com/atsign-foundation/at_server/pull/1471/commits/8ee7c6296c3707f9eeb1049140f66fa17e16d5dd). And the test config files and config util are part of [77c7be0](https://github.com/atsign-foundation/at_server/pull/1471/commits/77c7be013c407951bfe78a46650476131561b20d). What do you think about this approach @gkc

Reply:
Very good, thank you @srieteja